File: leetcode/#326-easy.cpp
/*
 * https://leetcode.com/problems/power-of-three/
 * 326. Power of Three
 */

class Solution {
public:
    bool isPowerOfThree(int n) {
        if(n <= 0) return false;
        
        while(n >= 3){
            if(n%3) return false;
            n /= 3;
        }
        return n == 1;
    }
};

File: leetcode/Algorithm/#326-easy.py
#
# https://leetcode.com/problems/power-of-three/
# 326. Power of Three
# 

class Solution(object):
    def isPowerOfThree(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n <= 0:
            return False
        return 3**19 % n == 0
        